virocon/distributions.py

```diff
@@ -1184,3 +1185,163 @@ def _wlsq_error(delta, x, p, w, return_alpha_beta=False, falpha=None, fbeta=None
         wlsq_error = np.sum(w * (x - x_hat) ** 2)
 
         return wlsq_error
+
+
+class GeneralizedGammaDistribution(Distribution):
+    """
+    A generalized Gamma distribution. 
+   
+    The distributions probability density function is given by [5]_ . 
+    The parametrization is orientated on [5]_ :
+    
+    :math:`f(x) = \\frac{ \\lambda^{cm} cx^{cm-1} \\exp \\left[ - \\left(\\lambda x^{c} \\right) \\right] }{\\Gamma(m)}`
+    
+    Parameters
+    ----------
+    m : float
+        Shape parameter of the generalized Gamma distribution. Defaults to 1.
+    c : float
+        Second shape parameter of the generalized Gamma distribution. Defaults
+        to 1.
+    lambda_ : float
+        Additional scale parameter of the generalized Gamma distribution 
+        (3-parameter generalized Gamma distribution). Defaults to 1.
+    f_m : float
+        Fixed shape parameter of the generalized Gamma distribution (e.g. 
+        given physical parameter). If this parameter is set, c is ignored. 
+        Defaults to None.
+    f_c : float
+       Fixed second shape parameter of the generalized Gamma distribution (e.g.
+       given  physical parameter). If this parameter is set, k is ignored. 
+       Defaults to None. 
+    f_lambda_ : float
+        Fixed scale parameter of the generalized Gamma distribution (e.g. 
+        given physical parameter). If this parameter is set, b is ignored. 
+        Defaults to None.
+
+    References
+    ----------
+    .. [5] M.K. Ochi, J.E. Wahlen, Prediction of severest significant wave 
+        height, Coast. Eng. Chap. 36 (1980) 587e599.
+        
+    """
+
+    def __init__(self, m=1, c=1, lambda_=1, f_m=None, f_c=None, f_lambda_=None):
+
+        # TODO set parameters to fixed values if provided
+        self.m = m  # shape
+        self.c = c  # shape
+        self.lambda_ = lambda_  # scale
+        self.f_m = f_m
+        self.f_c = f_c
+        self.f_lambda_ = f_lambda_
+
+    @property
+    def parameters(self):
+        return {"m": self.m, "c": self.c, "lambda": self.lambda_}
```

Comment:
Should this be "lambda" or maybe rather "lambda_"? I am not sure how we handled this in the other distributions.

Reply:
The other distributions do not add a _ to the parameter. To stay consistent, I would propose to stick to "lambda"?

Reply:
This brings up the question of which consistency to use. returning "lambda" is consistent with the other distributions but inconsistent with the lambda_ parameter itself. I would suggest we keep it as is for now, but discuss this in general in a new issue.

Reply:
As discussed, I changed it to lambda_ . It is consistent in the sense that we always write the parameter name as it is indicated in the description of the class. Just because the other parameters don't have an underscore doesn't change the fact that we should call the parameter the way we described it in the beginning.